### ISSUE TITLE
More contrast in text for themes

### ISSUE THREAD

**Issue by Fmstrat** [hjiangsu/thunder#223]
**Is your feature request related to a problem? Please describe.**
With content displayed in the post feed, it's all starting to run together:

![text](https://github.com/hjiangsu/thunder/assets/971474/e755618f-16f1-4b6b-8634-76bb1c6bcdb8)

**Describe the solution you'd like**
Updates to theming to support content adjustments, including expansion of themes to use a drop down with "System", "Dark", "Light" to be the defaults and "OLED" + others to be additional options.

**Describe alternatives you've considered**
Making content displayed in the feed gray instead of white. Realistically, this could work, but a broader theme capability would be nice.

**Comment by machinaeZER0** [comment]
Oh yeah - in compact view the non-title text all looks to be a slightly darker shade, which I agree is helpful for readability when scrolling!

**Comment by CTalvio** [comment]
The color on the community name and instance should probably match between compact and normal.

**Comment by acidrums4** [comment]
I think choosing a bold(er) and a bit bigger font for post titles, and a lighter and dimmer font for metadata (magazine and instance names) could work.

I prepared this sketch with Inkscape to show what I mean. Note that I used the "Manrope" typeface as I did not have Roboto installed on my computer.

![bitmap](https://github.com/hjiangsu/thunder/assets/513421/ac580081-1273-46ea-81ae-f838a420bbd1)

**Comment by machinaeZER0** [comment]
Yeah, this is pretty much exactly how compact mode handles font colors :)

**Comment by CTalvio** [comment]
Should I make this change or is someone else already on it?

**Comment by machinaeZER0** [comment]
This feels like it would be a total lay-up for me to try and tackle, but I've never done a PR before and I'm not sure where to begin :(

**Comment by CTalvio** [comment]
> This feels like it would be a total lay-up for me to try and tackle, but I've never done a PR before and I'm not sure where to begin :( presumably I'd find where the color values are set for normal and compact modes and then make them match?

Yes.

I'll get you started. The files are `lib/community/widgets/post_card_view_comfortable.dart` and `lib/community/widgets/post_card_view_compact.dart`.

**Comment by machinaeZER0** [comment]
Thank you! As mentioned over there, I don't actually know how to set up a build environment on Windows (or at all, honestly) to compile and test in Thunder myself, but looking at those two files side by side I believe I have been able to get compact and "normal" view to match with this PR?

Would be interested in setting up a test environment in the future! Would be cool to be able to help out in small ways such as this :)

**Comment by CTalvio** [comment]
It was not that difficult.

First: https://docs.flutter.dev/get-started/install
Then check out the project in your IDE of choice: for example [VScode](https://learn.microsoft.com/en-us/azure/developer/javascript/how-to/with-visual-studio-code/clone-github-repository?tabs=create-repo-command-palette%2Cinitialize-repo-activity-bar%2Ccreate-branch-command-palette%2Ccommit-changes-command-palette%2Cpush-command-palette)

I'm using Android Studio on linux, as I've used it before, years ago.

**Comment by hjiangsu** [comment]
> Thank you! As mentioned over there, I don't actually know how to set up a build environment on Windows (or at all, honestly) to compile and test in Thunder myself, but looking at those two files side by side I believe I have been able to get compact and "normal" view to match with this PR?
> 
> 
> 
> Would be interested in setting up a test environment in the future! Would be cool to be able to help out in small ways such as this :)

Hey, thanks for your interest in helping out! I think the hardest part in all of this is just setting up the environment. The link that CTalvio sent should be the best way to learn how to set everything up

If you do have questions on all this, feel free to join in the Matrix space and I can also guide you on the setup process: https://matrix.to/#/%23thunderapp%3Amatrix.org

Once you have the environment all set up, you'll have to learn a few commands for git, which is what is used to help with "pushing" or "committing" the code changes you've made :D

**Comment by machinaeZER0** [comment]
I thinkkkkk I have successfully created a pull request for this now, haha. Based my code off the dev branch this time

**Comment by machinaeZER0** [comment]
@hjiangsu I think this can potentially be marked as fixed in the next release?

**Comment by hjiangsu** [comment]
I'll temporarily marked this as fixed in the next release - however, I feel like this issue might encompass a few more things, so let me know if there's anything missing here and I'll remove the label

**Comment by Fmstrat** [comment]
> I'll temporarily marked this as fixed in the next release - however, I feel like this issue might encompass a few more things, so let me know if there's anything missing here and I'll remove the label

Is there a separate issue for multi-themes?

**Comment by machinaeZER0** [comment]
What's a multi-theme?

**Comment by hjiangsu** [comment]
If you mean having the ability to select multiple themes or adding new themes, I don't believe theres an issue created yet for that (but it has been in discussion in a few places for example https://github.com/hjiangsu/thunder/pull/234)

I think we can create a new issue for supporting multi-theming!
